src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets

```diff
@@ -33,14 +33,12 @@ Copyright (c) .NET Foundation. All rights reserved.
   <!-- Resolve the RazorLangVersion based on values imported or TFM. -->
   <PropertyGroup>
     <_TargetFrameworkVersionWithoutV>$(TargetFrameworkVersion.TrimStart('vV'))</_TargetFrameworkVersionWithoutV>
-
-    <!-- Workaround for an older build of DesignTime.targets that requires this property to be available. This can be removed after 16.1 preview 2 insertion -->
-    <_TargetingNETCoreApp30OrLater Condition="'$(_TargetFrameworkVersionWithoutV)' &gt; '2.9'">true</_TargetingNETCoreApp30OrLater>
+    <_TargetingNETCoreApp30OrLater Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(_TargetFrameworkVersionWithoutV)' >= '3.0'">true</_TargetingNETCoreApp30OrLater>
```

Comment:
This was missing a check for TargetFramework. Updated places in the code that used `_TargetFrameworkVersionWithoutV`  to use `_TargetingNETCoreApp30OrLater` which does the correct check. In addition, added tests for .NET Framework based 1.1 projects so we know we don't regress those either.